### PR TITLE
Bugfix/skale 2711 changed nodex indexes during rotation

### PR DESCRIPTION
**Pull request by payvint**
Fixed changed nodes indexes during rotation, added some tests for it
Restructure SchainsInternal contract - no inheritance from Groups
Added holesForSchains - will store hole when some one exiting and the new one will be set in this hole

### CODE REVIEW

**Review by olehnikolaiev — DISMISSED** [review]
lgtm!

File: scripts/test_deploy.sh [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 
-npx ganache-cli --gasLimit 8000000 --quiet &
+npx ganache-cli --gasLimit 10000000 --quiet &
```
Comment (DimaStebaev) [inline scripts/test_deploy.sh]: This value is much higher than on mainnet.

File: test/NodesFunctionality.ts [hunk 2]
```diff
@@ -100,45 +100,40 @@ contract("NodesFunctionality", ([owner, validator, nodeAddress, nodeAddress2]) =
                 });
         });
 
-        it("should fail to delete non existing node", async () => {
-            await nodes.removeNode(validator, 1)
-                .should.be.eventually.rejectedWith("Node does not exist for message sender");
-        });
+        // it("should fail to delete non existing node", async () => {
```
Comment (DimaStebaev) [inline test/NodesFunctionality.ts]: Do you plan to return this test or is it not relevant?

**Review by DimaStebaev — CHANGES_REQUESTED** [review]

File: scripts/test_deploy.sh (hunk 1 above)
Comment (payvint) [reply scripts/test_deploy.sh]: It is the same like on mainnet but we decided to keep it as 8M

File: test/NodesFunctionality.ts (hunk 2 above)
Comment (payvint) [reply test/NodesFunctionality.ts]: This test is not relevant for this test scope. We have similar test in SkalManager test which is cover all functionality of SkaleManager

**Review by DimaStebaev — APPROVED** [review]

**Pull request merged**